src/User/UserServiceProvider.php

```diff
@@ -61,6 +74,26 @@ protected function registerAvatarsFilesystem()
     public function boot()
     {
         $this->app->make('flarum.gate')->before(function (User $actor, $ability, $model = null) {
+            $evaluationCriteria = [
+                AbstractPolicy::$FORCE_DENY => false,
+                AbstractPolicy::$FORCE_ALLOW => true,
+                AbstractPolicy::$DENY => false,
+                AbstractPolicy::$ALLOW => true,
+            ];
+
+            $results = [];
+            foreach ($this->app->make('flarum.policies') as $policy) {
+                $results[] = $this->app->make($policy)->checkAbility($actor, $ability, $model);
```

Comment:
Before this refactor, when policies were listeners, they were resolved only once.

How about we resolve them all in an array before making the gate and then we use that array of all resolved policies inside.